node/src/components/consensus/protocols/highway.rs

```diff
@@ -320,6 +320,25 @@ where
         self.highway.has_evidence(vid)
     }
 
+    fn request_evidence(
+        &self,
+        sender: I,
+        vid: &C::ValidatorId,
+    ) -> Result<Vec<CpResult<I, C>>, Error> {
+        Ok(self
+            .highway
+            .validators()
+            .get_index(vid)
+            .and_then(|vidx| self.highway.get_dependency(&Dependency::Evidence(vidx)))
+            .map(|vv| {
+                let msg = HighwayMessage::NewVertex(vv.into());
+                let serialized_msg = rmp_serde::to_vec(&msg).expect("should serialize message");
+                ConsensusProtocolResult::CreatedTargetedMessage(serialized_msg, sender)
+            })
+            .into_iter()
+            .collect())
+    }
```

Comment:
We currently just make all of the still-bonded eras send evidence if they have any. There's not point making those that don't reply `Absent`. Also, if someone requests evidence from us they're either malfunctioning or they are requesting it because they know we have it.

Sending an `Absent` message if none of the still-bonded eras has evidence would complicate the code quite a bit, and then we'd need to add handling of the `Absent` message. So if we do that at all I'd make it a separate Jira issue (and have a discussion on whether we need it).